### PR TITLE
Add integration test for snapshot generation

### PR DESCRIPTION
**Pull request by allanmckenzie**

### CODE REVIEW

File: cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        databaseCleaner.cleanStreamBufferTable(DB_CONTEXT_NAME);
+        databaseCleaner.cleanStreamStatusTable(DB_CONTEXT_NAME);
+    }
+
+    private List<Event> createEvents(final long numberToCreate, final UUID streamId) {
+
+        final long firstPositionInStream = 1;
+        final ArrayList<Event> events = new ArrayList<>();
+        final Event recipeAddedEvent = createRecipeAddedEvent(firstPositionInStream, streamId);
+        events.add(recipeAddedEvent);
```
Comment (arcadius) [inline cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java:149]: we could have the loop below start with 1 and have this code moved into the loop?

File: cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    @Test
+    public void shouldUseJmxCommandToHydrateAggregateAndStoreSnapshot() throws Exception {
+
+        final String aggregateClassName = Recipe.class.getName();
+        final UUID recipeId = randomUUID();
+
+        snapshotJdbcRepository.removeAllSnapshots(recipeId, Recipe.class);
+        assertThat(snapshotJdbcRepository.getLatestSnapshot(recipeId, Recipe.class), is(empty()));
+
+        System.out.println("Generating " + NUMBER_OF_EVENTS_ON_STREAM_TO_CREATE + " events");
```
Comment (arcadius) [inline cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java:81]: maybe logger?

File: cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java (hunk 1 above)
Comment (allanmckenzie) [reply cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java:149]: They are different events :-( so I don't think that's possible. 
The first call creates a 'cakeshop.events.recipe-added' event, the subsequent calls create 'cakeshop.events.recipe-renamed' events

File: cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java (hunk 2 above)
Comment (allanmckenzie) [reply cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java:81]: It's very noisy to use a logger. Also, as the wildfly logging is sent to the console during cake-shop integration tests, it's hard to then see if the log message is a wildfly log message, or just information outputted by the test

**Pull request merged**